packages/block-editor/src/components/block-edit/index.js

```diff
@@ -12,7 +12,7 @@ import { Component } from '@wordpress/element';
  * Internal dependencies
  */
 import Edit from './edit';
-import { BlockEditContextProvider } from './context';
+import { BlockEditContextProvider, useBlockEditContext } from './context';
```

Comment:
I don't think we need the changes in this file, since you can access `clientId` from the context, you can call `updateBlockAttributes` and  `getBlockAttributes` without extra context values.

Reply:
Oh right. I was thinking of the Block API functions. I'll change it.

Reply:
It just feels a lot clunkier to have to use the context to get the `id` and then pass it to `useSelect`.

Don't you think a `useBlockEditContext` with attributes and the setter will be very useful for people building custom hooks that need access to attributes?

Reply:
I don't think so personally, I feel like the context should contain the minimum possible things because 
- these are new APIs
- semantically speaking, the context says, we're editing that block (id) and it should be enough for the consumer. 

if we add these, the question becomes: Will we add a new value there each time  we need another selector/action?

Reply:
>Will we add a new value there each time we need another selector/action?

I don't think we could, because it would be limited to the props that `BlockEdit` receives. I just tend to favor context over selector/subscription boilerplate, but I can see why this aligns more with the rest of the codebase and see the value in having a single way of doing things:

https://github.com/WordPress/gutenberg/pull/16781/commits/69865ddd252be9ea9c853174dbdc48ddf7b88c21